oci/oci.go

```diff
@@ -1 +1,91 @@
 package oci
+
+import (
+	"github.com/pkg/errors"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+// annotation spex from https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys
+const (
+	separator = `(?:[-._:@+]|--)`
+	alphanum  = `(?:[A-Za-z0-9]+)`
+	component = `(?:` + alphanum + `(?:` + separator + alphanum + `)*)`
+)
+
+var refRegexp = regexp.MustCompile(`^` + component + `(?:/` + component + `)*$`)
+var windowsRefRegexp = regexp.MustCompile(`^([a-zA-Z]:\\.+?):(.*)$`)
+
+// ValidImageName returns true is the specified image matches the open-containers image name specs, false otherwise.
+func ValidImageName(image string) bool {
+	return refRegexp.MatchString(image)
+}
+
+// SplitDirAndImage tries to split the provided OCI reference into the OCI directory and image.
+// The image is returned as string pointer, allowing for nil being returned, if no image part could be extraced from the reference.
+// Neither directory nor image parts are validated at this stage.
+func SplitDirAndImage(reference string) (string, *string) {
+	if runtime.GOOS == "windows" {
+		return splitDirAndImageWindows(reference)
+	}
+	return splitDirAndImageNonWindows(reference)
+}
+
+func splitDirAndImageWindows(reference string) (string, *string) {
+	groups := windowsRefRegexp.FindStringSubmatch(reference)
+	// nil group means no match
+	if groups == nil {
+		return reference, nil
+	}
+
+	// we expect three elements. First one full match, second the capture group for the dir and
+	// the third the capture group for the image
+	if len(groups) != 3 {
+		return reference, nil
+	}
+	return groups[1], &groups[2]
+}
+
+func splitDirAndImageNonWindows(reference string) (string, *string) {
+	sep := strings.SplitN(reference, ":", 2)
+	dir := sep[0]
+
+	var image *string
+	if len(sep) == 2 {
+		image = &sep[1]
+	}
+	return dir, image
+}
+
+// ValidateScope takes the OCI directory and validates its scope.
+func ValidateScope(scope string, dir string) error {
+	if runtime.GOOS == "windows" {
+		return validateScopeWindows(scope, dir)
+	}
+	return validateScopeNonWindows(scope, dir)
+}
+
+func validateScopeWindows(scope string, dir string) error {
+	matched, _ := regexp.Match(`^[a-zA-Z]:\\`, []byte(dir))
+	if !matched {
+		return errors.Errorf("Invalid scope '%s'. Must be an absolute path", dir)
+	}
+
+	return nil
+}
+
+func validateScopeNonWindows(scope string, dir string) error {
+	if !strings.HasPrefix(dir, "/") {
+		return errors.Errorf("Invalid scope %s: must be an absolute path", dir)
+	}
+
+	// Refuse also "/", otherwise "/" and "" would have the same semantics,
+	// and "" could be unexpectedly shadowed by the "/" entry.
+	// (Note: we do allow "/:someimage", a bit ridiculous but why refuse it?)
```

Comment:
The parenthesized note is no longer relevant now that identities/scopes do not contain the image name.

Reply:
I'll remove it